File: LeetCode/Graphs/797. All Paths From Source to Target.java
class Solution {
    public int target;
    public List<Integer> temp = new ArrayList<>();
    public List<List<Integer>> res = new ArrayList<>();

    public List<List<Integer>> allPathsSourceTarget(int[][] graph) {
        target = graph.length - 1;
        dfs(graph, 0);
        return res;    
    }

    public void dfs(int[][] graph, int parent) {
        temp.add(parent);
        if(parent == target) res.add(new ArrayList<Integer>(temp));
        else for(int i : graph[parent]) {
            dfs(graph, i);
        }
        temp.remove(temp.size() - 1);
    }
}

File: LeetCode/Graphs/797. All Paths From Source to Target.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int target;
    vector<vector<int>> res;
    vector<int> temp;

    void dfs(vector<vector<int>>& graph, int parent) {
        temp.push_back(parent);
        if(parent == target) res.push_back(temp);
        else for(auto c : graph[parent]) {
            dfs(graph, c);
        }
        temp.pop_back();
    }

    vector<vector<int>> allPathsSourceTarget(vector<vector<int>>& graph) {
        target = graph.size() - 1;
        dfs(graph, 0);
        return res;
    }
};